### PR TITLE
GROOVY-9951: set implicit-this to false for method call expressions

### PR DESCRIPTION
**Pull request by eric-milles**
https://issues.apache.org/jira/browse/GROOVY-9951

### CODE REVIEW

File: src/test/groovy/transform/stc/BugsSTCTest.groovy [hunk 1]
```diff
@@ -938,4 +938,22 @@ Printer
...
     }
+
+    // GROOVY-9951
+    void testInnerImmutablePOJO() {
+        assertScript '''
+            import groovy.transform.Immutable
+            import groovy.transform.stc.POJO
+
+            class Outer {
+                @Immutable @POJO
```
Comment (danielsun1106) [inline src/test/groovy/transform/stc/BugsSTCTest.groovy:949]: AFAIR, `@POJO` should be used with `@CompileStatic`.

Comment (eric-milles) [reply src/test/groovy/transform/stc/BugsSTCTest.groovy:949]: This test runs under STC and SC.

Comment (danielsun1106) [reply src/test/groovy/transform/stc/BugsSTCTest.groovy:949]: OK

File: src/main/java/org/apache/groovy/ast/tools/ConstructorNodeUtils.java [hunk 2]
```diff
@@ -88,17 +91,20 @@ public static Statement checkPropNamesS(final VariableExpression namedArgs, fina
...
+        MethodCallExpression names = callX(namedArgs, "keySet");
+        names.setImplicitThis(false);
+
+        MethodCallExpression isNameValid = callX(validNames, "contains", varX(name));
+        isNameValid.setImplicitThis(false);
+
+        return block(
+            declS(validNames, listX(props.stream().map(p -> constX(p.getName())).collect(toList()))),
+            forS(name, names, ifS(notX(isNameValid),
+                    throwS(ctorX(EXCEPTION, plusX(constX("Unknown named argument: "), varX(name))))))
```
Comment (paulk-asert) [inline src/main/java/org/apache/groovy/ast/tools/ConstructorNodeUtils.java:107]: This all LGTM but we can probably go one step further for the plusX expression if `pojo` is true. At the moment it ends up with a call to `StringGroovyMethods#plus` but we could do such a thing without needing the Groovy runtime (for the `pojo` case). There might be multiple ways to do that and we can do that as a next step.

**Comment by paulk-asert** [comment]
Merged, thanks!

**Pull request closed without merging**